### PR TITLE
Unique event name override

### PR DESCRIPTION
**Pull request by konraddysput**
# Why

This diff adds an option to override default unique event name. In some systems we have a different attributes responsible for generating unique user attribute that we should use on the service side

### CODE REVIEW

File: backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java [hunk 1]
```diff
@@ -117,18 +124,40 @@ public final class BacktraceMetrics implements Metrics {
...
      * Enable metrics
      *
      * @param settings for Backtrace metrics
      */
     public void enable(BacktraceMetricsSettings settings) {
+        enable(settings, defaultUniqueEventName);
+    }
+
+    public void enable(BacktraceMetricsSettings settings, String uniqueEventName) {
+        if(uniqueEventName == null || uniqueEventName.length() == 0) {
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java:157]: fix formatting

File: example-app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -46,7 +46,7 @@
         android:id="@+id/dumpWithoutCrash"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
-        android:text="DumpWithoutCrash"
+        android:text="Generate sample crash report"
```
Comment (BartoszLitwiniuk) [inline example-app/src/main/res/layout/activity_main.xml:49]: why?

File: backtrace-library/src/main/java/backtraceio/library/BacktraceCredentials.java [hunk 3]
```diff
@@ -124,7 +124,7 @@ public String getSubmissionToken() {
         final String tokenQueryParam = "token=";
         String submissionUrl = getSubmissionUrl().toString();
         String token = submissionUrl.contains("submit.backtrace.io")
-                ? submissionUrl.substring(submissionUrl.lastIndexOf("/") - tokenLength + 1, submissionUrl.lastIndexOf("/"))
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/main/java/backtraceio/library/BacktraceCredentials.java:127]: why? could you add/fix unit test?

File: example-app/src/main/res/layout/activity_main.xml (hunk 2 above)
Comment (konraddysput) [reply example-app/src/main/res/layout/activity_main.xml:49]: DumpWIthoutCrash is our internal interface name. It doesn't tall the user what it does. "Generate sample crash report" tells exactly what user can expect by pressing the button

**Comment by vlussenburg** [comment]
Unit test failure!
org.junit.ComparisonFailure: expected:<[]aaaaabbbbbccccf82668...> but was:<[/]aaaaabbbbbccccf82668...>

File: backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java (hunk 1 above)
Comment (konraddysput) [reply backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java]: doe

File: backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java [hunk 4]
```diff
@@ -1,12 +1,15 @@
 package backtraceio.library.services;
 
 import android.content.Context;
+
 import androidx.annotation.NonNull;
 
+import java.security.InvalidParameterException;
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java:7]: is it used?

File: backtrace-library/src/main/java/backtraceio/library/BacktraceCredentials.java [hunk 5]
```diff
@@ -123,10 +123,10 @@ public String getSubmissionToken() {
         final int tokenLength = 64;
         final String tokenQueryParam = "token=";
         String submissionUrl = getSubmissionUrl().toString();
-        String token = submissionUrl.contains("submit.backtrace.io")
-                ? submissionUrl.substring(submissionUrl.lastIndexOf("/") - tokenLength + 1, submissionUrl.lastIndexOf("/"))
-                : submissionUrl.substring(submissionUrl.indexOf(tokenQueryParam) + tokenQueryParam.length(),
-                    submissionUrl.indexOf(tokenQueryParam) + tokenQueryParam.length() + tokenLength - 1);
-        return token;
+        if(submissionUrl.contains("submit.backtrace.io")) {
```
Comment (BartoszLitwiniuk) [inline backtrace-library/src/main/java/backtraceio/library/BacktraceCredentials.java:126]: fix formatting

File: backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java (hunk 4 above)
Comment (konraddysput) [reply backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java:7]: Yup

File: backtrace-library/src/main/java/backtraceio/library/BacktraceCredentials.java (hunk 5 above)
Comment (konraddysput) [reply backtrace-library/src/main/java/backtraceio/library/BacktraceCredentials.java]: done

**Review by BartoszLitwiniuk — APPROVED** [review]
lgtm

File: backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java (hunk 4 above)
Comment (BartoszLitwiniuk) [reply backtrace-library/src/main/java/backtraceio/library/services/BacktraceMetrics.java:7]: where? 
image

**Review by rick-bt — APPROVED** [review]
lgtm2

**Pull request merged**